### PR TITLE
Incomplete driver code for alc5658

### PR DESCRIPTION
**Pull request by btheosam**

### CODE REVIEW

File: os/arch/arm/src/s5j/chip/s5jt200_pinconfig.h [hunk 1]
```diff
@@ -160,6 +162,7 @@
 #define GPIO_I2S0_BCLK     (GPIO_ALT1 | GPIO_FLOAT | GPIO_FAST1X | GPIO_PORTG3 | GPIO_PIN0)
 #define GPIO_I2S0_LRCK     (GPIO_ALT1 | GPIO_FLOAT | GPIO_FAST1X | GPIO_PORTG3 | GPIO_PIN1)
 #define GPIO_I2S0_SDI      (GPIO_ALT1 | GPIO_FLOAT | GPIO_FAST1X | GPIO_PORTG3 | GPIO_PIN2)
+#define GPIO_I2S0_SDO      (GPIO_ALT1 | GPIO_FLOAT | GPIO_FAST1X | GPIO_PORTG3 | GPIO_PIN2)
```
Comment (jsdosa) [inline os/arch/arm/src/s5j/chip/s5jt200_pinconfig.h]: is this correct?
SDI is the same as Define, right?

Comment (btheosam) [inline os/arch/arm/src/s5j/chip/s5jt200_pinconfig.h]: Yes, based on configuration, same pin will work as IN/OUT.

File: os/arch/arm/src/sidk_s5jt200/include/s5jt200_alc5658_i2c.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/****************************************************************************
```
Comment (Taejun-Kwon) [inline os/arch/arm/src/sidk_s5jt200/include/s5jt200_alc5658_i2c.h]: I think it doesn't have any dependency with s5jt200 chip

File: os/arch/arm/src/sidk_s5jt200/src/s5jt200_alc5658.c [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+		i2c = up_i2cinitialize(CONFIG_ALC5658_I2C_PORT);
+		if (!i2c) {
+			auderr("ERROR: Failed to initialize CODEC I2C%d\n");
+			ret = -ENODEV;
+			goto errout;
+		}
+
+		/* Get an instance of the I2S interface for the ALC5658 data channel */
+
+		i2s = (struct i2s_dev_s *) s5j_i2s_initialize();
```
Comment (Taejun-Kwon) [inline os/arch/arm/src/sidk_s5jt200/src/s5jt200_alc5658.c]: Like up_i2cinitialize, i2s also needs interface.
Defineing i2s in common header, and each codec chip should implements interface.

**Comment by btheosam** [comment]
This changes are merged part of another commit.

**Pull request closed without merging**